lib/spack/spack/binary_distribution.py

```diff
@@ -895,13 +958,17 @@ def generate_package_index(cache_prefix):
     )
 
     try:
-        _read_specs_and_push_index(file_list, cache_prefix, db, db_root_dir)
+        _read_specs_and_push_index(
+            file_list, read_method, cache_prefix, db, db_root_dir, concurrency
+        )
     except Exception as err:
         msg = "Encountered problem pushing package index to {0}: {1}".format(cache_prefix, err)
         tty.warn(msg)
         tty.debug("\n" + traceback.format_exc())
     finally:
         shutil.rmtree(tmpdir)
+        if tmpspecsdir:
+            shutil.rmtree(tmpspecsdir)
```

Comment:
This is needed only if we use `aws`. It would be better if we can scope it with that "reading strategy", not in a higher level function.